misk/src/test/kotlin/misk/client/HttpClientEnvoyTest.kt

```diff
@@ -0,0 +1,93 @@
+package misk.client
+
+import com.google.inject.Guice
+import com.google.inject.Injector
+import com.google.inject.Provides
+import com.google.inject.name.Names
+import helpers.protos.Dinosaur
+import misk.MiskServiceModule
+import misk.inject.KAbstractModule
+import misk.inject.getInstance
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import retrofit2.Call
+import retrofit2.http.Body
+import retrofit2.http.POST
+import java.io.File
+import javax.inject.Singleton
+
+/** Demonstrates ability to bind up an envoy-based http client and connect via unix sockets. */
+internal class HttpClientEnvoyTest {
+  private lateinit var socketFile: File
+  private lateinit var server: MockWebServer
+  private lateinit var clientInjector: Injector
+
+  @BeforeEach
+  fun createClientAndServer() {
+    socketFile = File("/tmp/ClientAndServer.sock")
+    server = MockWebServer()
+    server.setServerSocketFactory(UnixDomainServerSocketFactory(socketFile))
+
+    clientInjector = Guice.createInjector(
```

Comment:
why not just use a `MiskTest` and let it do the injection for you?

Reply:
also you could mode the `before` and `after` as a `Service` which is the misky way

Reply:
done